content/ytt/docs/latest/data-values-schema-migration-guide.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Schema Migration Guide
+toc: "true"
+---
+
+Schema documents provide a way to declare data values with their types, and default values. Without schemas, validating the presence of data values requires additional `ytt` configuration containing starlark assertions.
+
+[Learn more about using schema here.](ytt-schema.md)\
+[Read the detailed schema module here.](lang-ref-ytt-schema.md)
+
+
+## How do I, a configuration author, migrate my `ytt` library to use schemas?
+
+To make use of the schema feature, your `ytt` invocation must first contain files using the [data values feature](ytt-data-values.md). Migrating to schemas involves converting your data values files into a schema file.
+
+### Single data values file
+
+Starting with a single data values file, `values.yml`:
+```yaml
+#@data/values
+---
+key1: myVal
+key2: 8080
+```
+
+Convert this data values file into schema by changing the top level annotation in the document to say `#@data/values-schema`, and (optional) rename `values.yml` to `values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8080
+```
+Now simply include this schema file in your `ytt` invocation to receive the benefits of `ytt` schemas.
+
+
+**Note: If your data values file contains arrays (ie. `["example"]`, `- example`), be sure to [provide default values for arrays](#how-do-i-provide-default-values-for-an-array).**
+
+### Multiple data values files
+For this case, be sure that you are [using multiple data values files](ytt-data-values.md#splitting-data-values-into-multiple-files). 
+
+Given a `ytt` configuration with two data values files:
+```bash
+$ tree .
+.
+├── config.yml
+├── values-1.yml
+└── values-2.yml
+```
+
+`values-1.yml`:
+```yaml
+#@data/values
+---
+key1: myVal
+key2: 8080
+```
+
+`values-2.yml`:
+```yaml
+#@data/values
+---
+key2: 8088
+#@overlay/match missing_ok=True
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+The first step is to combine these two sets of data values into a single set of data values (a union set of all top level keys from the data values files).
+`ytt` can do this for you by providing the`--data-values-inspect` flag to your usual `ytt` command:
+```bash
+$ `ytt -f . --data-values-inspect
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+You can now copy this output into a yaml file and make it a schema by adding the `#@data/values-schema` annotation to the top of a new document.
+
+`values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+
+Now just include the `values-schema.yml` file in your `ytt` invocation instead of the multiple data values files.
+
+### Multiple data values files + Private libraries
+In this scenario, we expect your configuration to depend on a ytt library; outlined in the [library module docs](lang-ref-ytt-library.md).
```

Comment:
Similar as above: that it's "friendlier" to describe the circumstances from the _reader's_ perspective...

```suggestion
If your configuration depends on a ytt library — outlined in the [library module docs](lang-ref-ytt-library.md) — there are a few points to note.
```